internal/node/reverse_proxy/proxy.go

```diff
@@ -64,6 +64,7 @@ type RuleHandler interface {
 	Type() string
 	Match(url *url.URL) bool
 	Handler() http.Handler
+	Rank() int
```

Comment:
what do we need this for? I thought this could be implemented more natively, with the ordering in which we add rules?

Reply:
This was a quick hack. I think the way the reverse proxy is configured right now is very clunky, and I'm thinking about refactoring it. Can we defer until then?